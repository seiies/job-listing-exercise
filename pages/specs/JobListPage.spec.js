/* global jest, describe, it, expect */

import React from 'react'
import { JobListPage } from '../index'
import { shallow } from 'enzyme'

describe('index page', () => {
  let state
  const store = {
    getState: jest.fn(() => state),
    dispatch: jest.fn()
  }

  it('renders correctly', () => {
    let jobList = [
      {
        id: 1,
        image: 'https://place.cat/c/100/100',
        company: 'Deutsche Bahn AG',
        title: 'Job title'
      },
      {
        id: 2,
        image: 'https://place.cat/c/100/100',
        company: 'Deutsche Bahn AG',
        title: 'Job title'
      }
    ]

    const component = shallow(<JobListPage jobList={jobList} />)
    expect(component).toMatchSnapshot()
  })

  it('does not fetch job list in case of it is present in store', async () => {
    state = { jobList: ['something'] }

    await JobListPage.getInitialProps({ store })
    expect(store.dispatch).not.toBeCalled()
  })

  it('fetches job list in case of lack of it in store', async () => {
    state = { jobList: [] }

    await JobListPage.getInitialProps({ store })
    expect(store.dispatch).toHaveBeenCalledTimes(1)
  })
})
