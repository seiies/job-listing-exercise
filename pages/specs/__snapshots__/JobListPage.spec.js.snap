// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index page renders correctly 1`] = `
<React.Fragment>
  <SideEffect(Head)>
    <title>
      Job listings
    </title>
  </SideEffect(Head)>
  <Component>
    <h1>
      Job listings
    </h1>
    <JobList
      jobs={
        Array [
          Object {
            "company": "Deutsche Bahn AG",
            "id": 1,
            "image": "https://place.cat/c/100/100",
            "title": "Job title",
          },
          Object {
            "company": "Deutsche Bahn AG",
            "id": 2,
            "image": "https://place.cat/c/100/100",
            "title": "Job title",
          },
        ]
      }
    />
  </Component>
</React.Fragment>
`;
